src/pages/loginPage.tsx

```diff
@@ -0,0 +1,98 @@
+/* eslint-disable react-hooks/rules-of-hooks */
+import React, {useState} from 'react';
+import {
+  View,
+  Text,
+  TextInput,
+  Button,
+  Alert,
+  ActivityIndicator,
+  KeyboardAvoidingView,
+  Platform,
+} from 'react-native';
+import {loginAccess} from '../service/client';
+import {Navigation} from 'react-native-navigation';
+import styles from '../styles/loginPageStyles';
+
+interface PageProps {
+  componentId: string;
+  rootTag: number;
+}
+
+const loginPage = (props: PageProps) => {
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [loading, setLoading] = useState(false);
+
+  function inputValue() {
+    if (!passwordTest()) {
+      Alert.alert('Incorrect password format');
+    } else if (!emailTest()) {
+      Alert.alert('Incorrect email format');
+    } else {
+      onLoading();
+    }
+  }
+
+  function passwordTest() {
+    const passwordValidation = /(?=.{7,})(?=.*[0-9])(?=.*[a-z])|(?=.{7,})(?=.*[0-9])(?=.*[A-Z])/;
+    return passwordValidation.test(password) ? true : false;
+  }
+
+  function emailTest() {
+    return email.indexOf('@') && email.indexOf('.com') !== -1 ? true : false;
+  }
+
+  async function onLoading() {
```

Comment:
ainda nas convenções que a gente usa, a gente costuma chamar essas funções que fazem algo quando um evento acontece (loading, click, success/error e etc..) de handle<event>() entende? Então ficaria assim:
```suggestion
  async function handleLoading() {
```

Reply:
Sobre isso, acho que a função que tem que ter um `handle` de prefixo é a função  q vc chamou de "inputValue", porque ela vem efetivamente do clique do botão. Talvez essa função aqui possa ter uma abordagem imperativa mesmo, ou seja, ela pode ter o nome que reflete o que ela faz. No caso, ela faz o login e navega mesmo, não sei se `onLoading` é um bom nome né?

Sobre o `inputValue`, a abordagem declarativa seria, em vez de o nome da função descrever o que ela faz, ela descreve na verdade o evento que a trigga. No caso, acho `handleButtonTap` ou `handleSubmit` são nomes melhores.

Reply:
Humm. Faze sentido! Vou pensar em algo melhor então. Obrigado pelas dicas 😄